upload/lambdas/health_check/health_check.py

```diff
@@ -0,0 +1,109 @@
+from datetime import datetime, timedelta
+import json
+import os
+
+import boto3
+import requests
+
+from upload.common.database import _run_query
+from upload.common.logging import get_logger
+from upload.common.upload_config import UploadConfig
+
+logger = get_logger(__name__)
+
+client = boto3.client('cloudwatch')
+
+
+class HealthCheck:
+    def __init__(self):
+        self.env = os.environ['DEPLOYMENT_STAGE']
+        logger.debug(f"Running a health check for {self.env}. Results will be posted in #upload-service")
+        self.webhook = UploadConfig().slack_webhook
+
+        self.checksumming_areas_count_query = "SELECT COUNT(*) FROM checksum " \
+                                              "WHERE status='CHECKSUMMING' " \
+                                              "AND created_at > CURRENT_DATE - interval '4 weeks' " \
+                                              "AND updated_at > CURRENT_TIMESTAMP - interval '2 hours'"
+        self.validation_areas_count_query = "SELECT COUNT(*) FROM validation " \
+                                            "WHERE status='VALIDATING' " \
+                                            "AND created_at > CURRENT_DATE - interval '4 weeks' " \
+                                            "AND updated_at > CURRENT_TIMESTAMP - interval '2 hours'"
+
+        self.undeleted_areas_count_query = "SELECT COUNT(*) FROM upload_area " \
+                                           "WHERE created_at > CURRENT_DATE - interval '4 weeks' " \
+                                           "AND status != 'DELETED'"
+
+    def health_check(self):
+        deadletter_results = self.get_deadletter_count()
+        undeleted_upload_area_count = self.parse_and_query_db(self.undeleted_areas_count_query)
+        checksumming_areas = self.parse_and_query_db(self.checksumming_areas_count_query)
+        validating_areas = self.parse_and_query_db(self.validation_areas_count_query)
+        status_info = f"DEADLETTER_QUEUE: {int(deadletter_results['ApproximateNumberOfMessagesVisible'])} in queue, " \
+                      f"{int(deadletter_results['NumberOfMessagesReceived'])} added in past 24 hrs\n" \
+                      f"UPLOAD_AREAS: {undeleted_upload_area_count} undeleted areas, {checksumming_areas} stuck in " \
+                      f"checksumming, {validating_areas} stuck in validation"
+
+        attachments = [{
+            "title": f"Health Check Report for {self.env}:",
+            "color": "good",
+            "text": status_info
+        }]
+
```

Comment:
Could we possibly add # of upload-api and checksum-daemon lambda failures from the past 24 hours? Can probably fetch this from cloudwatch metrics?

Reply:
I'd suggest maybe making this a separate ticket.  Let's get this lambda into service with some minimal functionality, then add to it later.

Reply:
👍